### PR TITLE
Handle tenant-host application in system upgrader 

### PR DESCRIPTION
**Pull request by freva**
This will upgrade the new `tenant-host` application (defined in #9668) from the controller. This is a migration, so for a period controller will either:
1. Upgrade both tenant `host`s & `proxy` by deploying zone-app
2. Upgrade `proxy` by deploying zone-app and upgrade tenant host by setting target version on `host`
3. Upgrade tenant host by setting target version on `host` and not deploy zone-app at all as the zone has no `proxy` nodes.

These cases are handled as follows:
1. Controller will set the target version for `host`, but that has no effect. After all `config` and `proxyhost`s have upgraded, controller will deploy zone-app which will actually update the wanted version both `proxy` and `host`. Once upgraded, controller will be satisfied that both the `proxy` and the `tenantHost` applications have upgraded.
2. After setting the target version, tenant hosts will quickly get their wanted version set and start upgrading. When it deploys the zone-app, only `proxy` nodes will have their wanted version updated.
3. Tenant hosts upgrade as intended through the target version. The zone-app should never be deployed due to the changes in 8b97b04 (If there are no `proxy` nodes in the zone, it is already converged?)

Please verify my reasoning/assumptions, the last one is particularly difficult to test.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/SystemApplication.java [hunk 1]
```diff
@@ -23,43 +21,36 @@
 public enum SystemApplication {
 
     configServerHost(ApplicationId.from("hosted-vespa", "configserver-host", "default"), NodeType.confighost),
+    configServer(ApplicationId.from("hosted-vespa", "proxy-config-servers", "default"), NodeType.config),
```
Comment (hakonhall) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/SystemApplication.java]: proxy -> zone.

Comment (freva) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/SystemApplication.java]: Nice! Fixed.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/SystemUpgrader.java [hunk 2]
```diff
@@ -40,9 +40,12 @@ protected void upgrade(Version target, SystemApplication application, ZoneId zon
 
     @Override
     protected boolean convergedOn(Version target, SystemApplication application, ZoneId zone) {
-        return    minVersion(zone, application, Node::currentVersion).map(target::equals)
-                                                                     .orElse(true)
-               && application.configConvergedIn(zone, controller(), Optional.of(target));
+        Optional<Version> minVersion = minVersion(zone, application, Node::currentVersion);
+        // Skip application convergence check if there are no nodes belonging to the application in the zone
+        if (minVersion.isEmpty()) return true;
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/SystemUpgrader.java]: Should add a test for this scenario, which I assume is "system without proxy app"?

**Comment by mpolden** [comment]
Please consider my comment. Nice change overall 👍

**Review by mpolden — APPROVED** [review]

**Pull request merged**